docs/configuration.md

```diff
@@ -154,15 +154,16 @@ spec:
 | `vxlanPort`             | The UDP port for VXLAN (default: `4789`).                                                                                                                                                                                                                                                                                                                                                                  |
 | `vxlanVNI`              | The virtual network ID for VXLAN (default: `4096`).                                                                                                                                                                                                                                                                                                                                                        |
 | `mtu`                   | MTU for overlay network (default: `0`, which causes Calico to detect optimal MTU during bootstrap).                                                                                                                                                                                                                                                                                                        |
 | `wireguard`             | Enable wireguard-based encryption (default: `false`). Your host system must be wireguard ready (refer to the [Calico documentation](https://docs.projectcalico.org/security/encrypt-cluster-pod-traffic) for details).                                                                                                                                                                                     |
 | `flexVolumeDriverPath`  | The host path for Calicos flex-volume-driver(default: `/usr/libexec/k0s/kubelet-plugins/volume/exec/nodeagent~uds`). Change this path only if the default path is unwriteable (refer to [Project Calico Issue #2712](https://github.com/projectcalico/calico/issues/2712) for details). Ideally, you will pair this option with a custom ``volumePluginDir`` in the profile you use for your worker nodes. |
 | `ipAutodetectionMethod` | Use to force Calico to pick up the interface for pod network inter-node routing (default: `""`, meaning not set, so that Calico will instead use its defaults). For more information, refer to the [Calico documentation](https://docs.projectcalico.org/reference/node/configuration#ip-autodetection-methods).                                                                                           |
+| `envVars`               | Map of key-values (strings) for any calico-node [environment variable](https://docs.projectcalico.org/reference/node/configuration#ip-autodetection-methods).                                                                                                                                                                                                                                              |
```

Comment:
Would make sense to add some notes about the allowed format. Something along the lines of "The values must be suitable to be used as Container Environment Variables". Might be quite obvious, but still. Would be cool if we could add some link to upstream docs... Unfortunately I failed to find some comprehensive docs about this real quick. The comments on the k8s API type seem a bit too technical and detailed: https://github.com/kubernetes/kubernetes/blob/v1.25.3/staging/src/k8s.io/api/core/v1/types.go#L1988-L2003

Ah, there's also variable name expansion involved...

Reply:
I added a comment, that all vars must be strings. I believe that people in general understand how env vars work and how to use them.

Reply:
Env var names in Kubernetes are constrained to `^[-._a-zA-Z][-._a-zA-Z0-9]*$`, for example.